### DISCUSSION TITLE
Passing material reference to blueprints

### DISCUSSION THREAD

**Discussion by hefujie1988** [nxrighthere/UnrealCLR#60] [Help] [answered]
hello, i am learning UnrealCLR and i think it is very good thans to you. i find that it doesn't work when i call a SetMaterial fuction. Maybe i was wrong in  somewhere. 
thank you!

![image](https://user-images.githubusercontent.com/26532399/97786489-e3033b80-1be6-11eb-881e-fd84faf5cd02.png)
![image](https://user-images.githubusercontent.com/26532399/97786521-1b0a7e80-1be7-11eb-8b2b-fe0858112355.png)
![image](https://user-images.githubusercontent.com/26532399/97786905-9c631080-1be9-11eb-8a9e-7289c480b163.png)
![image](https://user-images.githubusercontent.com/26532399/97786545-44c3a580-1be7-11eb-950e-16a146ccb33a.png)

**Comment by nxrighthere** [comment, marked as answer]
Invocation of blueprint functions and events is based on strings with further parsing by the engine. A managed reference can't be passed to blueprints like that, instead, you can pass a path to the material.

**Reply by hefujie1988** [reply]
yeah,thank you very much!
